src/shared/api/dto.ts

```diff
@@ -1,183 +1,228 @@
-/* eslint-disable no-unused-vars */
-/* eslint-disable camelcase */
-import type { JsonObject } from 'type-fest';
+import { z } from 'zod';
+import BigNumber from 'bignumber.js';
+import invariant from 'tiny-invariant';
+
+const paginationSchema = z.object({
+  page: z.number(),
+  per_page: z.number(),
+  // TODO: remove nullable when backend is ready
+  total_pages: z.number().nullable(),
+  total_items: z.number().nullable(),
+});
+
+export type Pagination = z.infer<typeof paginationSchema>;
+
+export const paginatedSchema = <T extends z.ZodType>(item: T) =>
+  z.object({
+    pagination: paginationSchema,
+    items: item.array(),
+  });
+
+export interface Paginated<T> {
+  pagination: Pagination
+  items: T[]
+}
 
-export {};
+const paginationParamsSchema = z
+  .object({
+    page: z.number(),
+    per_page: z.number(),
+  })
+  .partial();
 
-declare global {
-  interface Tagged<T, Content> {
-    t: T
-    c: Content
-  }
+export type PaginationParams = z.infer<typeof paginationParamsSchema>;
 
-  interface PaginationParams {
-    page: number
-    page_size: number
-  }
+const metadataSchema = z.record(z.string(), z.any());
+
+const accountIdSchema = z.string().brand('AccountId');
+const domainIdSchema = z.string().brand('DomainId');
 
-  interface Paginated<T> {
-    pagination: {
-      page: number
-      page_size: number
-      total: number
-    }
-    data: T[]
+class AssetDefinitionId {
+  private assetName = '';
+  private assetDomain = '';
+
+  public constructor(name: string, domain: string) {
+    this.assetName = name;
+    this.assetDomain = domain;
   }
 
-  interface Asset {
-    account_id: string
-    definition_id: string
-    value: AssetValue
+  public get name() {
+    return this.assetName;
+  }
+  public get domain() {
+    return this.assetDomain;
+  }
+  public toString() {
+    return this.assetName + '#' + this.assetDomain;
   }
+}
 
-  type AssetValue =
-    | Tagged<'Quantity', string>
-    | Tagged<'BigQuantity', string>
-    | Tagged<'Fixed', string>
-    | Tagged<'Store', JsonObject>;
+export function transformToAssetDefinitionId(assetDefinitionId: string): AssetDefinitionId {
+  const parts = assetDefinitionId.split('#');
+  invariant(parts.length === 2, 'Invalid assetDefinitionId format');
 
-  type AssetValueType = 'Quantity' | 'BigQuantity' | 'Fixed' | 'Store';
+  const [asset, asset_domain] = parts;
 
-  interface AssetDefinition {
-    id: string
-    value_type: AssetValueType
-    mintable: Mintable
-  }
+  return new AssetDefinitionId(asset, asset_domain);
+}
 
-  type Mintable = 'Once' | 'Infinitely' | 'Not';
+class AssetId {
+  private assetName = '';
+  private assetDomain = '';
+  private accountSignatory = '';
+  private accountDomain = '';
```

Comment:
```suggestion
  private account: AccountId;
  private definition: AssetDefinitionId;
```

where both are also classes.